src/core/layout/qgslayoutitempolyline.cpp

```diff
@@ -344,6 +339,22 @@ QPainterPath QgsLayoutItemPolyline::shape() const
   return strokedOutline;
 }
 
+bool QgsLayoutItemPolyline::isValid() const
+{
+  QList<QPointF> uniquePoints;
```

Comment:
Maybe a little comment to ease readability here to tell that the polyline is valid if there is at least 2 different points

Reply:
Done